hotstuff/src/committer.rs

```diff
@@ -0,0 +1,269 @@
+use crate::consensus::{Round, CHANNEL_CAPACITY};
+use crate::error::{ConsensusError, ConsensusResult};
+use config::Committee;
+use crypto::Hash as _;
+use crypto::{Digest, PublicKey};
+use futures::future::try_join_all;
+use futures::stream::FuturesOrdered;
+use futures::stream::StreamExt as _;
+use log::{debug, error, info, log_enabled};
+use primary::Certificate;
+use std::cmp::max;
+use std::collections::{HashMap, HashSet};
+use store::Store;
+use tokio::sync::mpsc::{channel, Receiver, Sender};
+
+/// The representation of the DAG in memory.
+type Dag = HashMap<Round, HashMap<PublicKey, (Digest, Certificate)>>;
+
+/// The state that needs to be persisted for crash-recovery.
+struct State {
+    /// The last committed round.
+    last_committed_round: Round,
+    // Keeps the last committed round for each authority. This map is used to clean up the dag and
+    // ensure we don't commit twice the same certificate.
+    last_committed: HashMap<PublicKey, Round>,
```

Comment:
Why do you need the last committed round of other authorities? We all commit in sync as HS instructs us to